Libraries/Lists/__tests__/VirtualizedList-test.js

```diff
@@ -274,3 +274,118 @@ describe('VirtualizedList', () => {
     );
   });
 });
+
+describe('VirtualizedList > OnEndReached', () => {
+  const ITEM_HEIGHT = 100;
+  const APPENDED_ITEM_COUNT = 10;
+
+  let listItems, onEndReached, instance;
+
+  beforeEach(() => {
+    listItems = appendNewItems([], 10);
+
+    onEndReached = jest.fn(function() {
+      appendNewItems(listItems, APPENDED_ITEM_COUNT);
+    });
+
+    instance = createComponentInstance();
+  });
+
+  it('should not be called after initial rendering', () => {
+    const scroll = createScrollMethod();
+    scroll(0);
+
+    expect(onEndReached).not.toHaveBeenCalled();
+    expect(listItems.length).toBe(10);
+  });
+
+  it('should be called once after scrolling by 200', () => {
+    const scroll = createScrollMethod();
+    scroll(200);
+
+    expect(onEndReached).toHaveBeenCalledTimes(1);
+    expect(onEndReached).toHaveBeenLastCalledWith({
+      distanceFromEnd: 200,
+    });
+    expect(listItems.length).toBe(20);
+  });
+
+  it('should not be called twice in a short period while scrolling fast', () => {
```

Comment:
Same is true here - if you simply reduce `onEndReachedThreshold` to just 1.5 instead of the default of 2 (or tweak how much your test scrolls or renders) this test will pass with no changes to `VirtualizedList`.

Reply:
The expected behavior is that `onEndReached` will not be called in duplicate while scrolling if we really don't need additional data.

I don't think it's easy to find out the suitable `onEndReachedThreshold` value to match the number and layout of data used. Changing the value each time we use FlatList makes maintenance difficult and increases the chance of mistakes. I'm sure it's better to provide the most common defaults and work in general without any changes.